### PR TITLE
[Docs] fix docs

### PR DESCRIPTION
**Pull request by Z-Fran**
1. config, datasets and visualization of user_guides
2. advanced_guides/transforms.md
3. advanced_guides/dataset.md
4. advanced_guides/losses.md
5. migration.md
6. model_zoo.md
7. notes: contribution, faq, projects

### CODE REVIEW

File: docs/en/advanced_guides/dataset.md [hunk 1]
```diff
@@ -2,85 +2,32 @@
...
-  Vimeo90K dataset for video super resolution for recurrent networks.
-
-### Video Frame Interpolation
-
-- [VFIVimeo90KDataset](/mmedit/datasets/vfi_vimeo90k_dataset.py)
-  Vimeo90K dataset for video frame interpolation.
-
-### Matting
-
+- [BasicImageDataset](/mmedit/datasets/basic_image_dataset.py)
```
Comment (plyfager) [inline docs/en/advanced_guides/dataset.md:5]: List basic datasets without introduction seems abrupt. Maybe you can say: We support three types of data, image, frames, xxx, and customized dataset are supposed to inherit from one of them xxx.

**Review by plyfager — COMMENTED** [review]
Datasets between two versions differ a lot, like ``evaluate`` is not supposed to appear in dataset. These changes should be reflected on docs. Maybe a overall check is needed.

File: docs/en/advanced_guides/dataset.md [hunk 2]
```diff
@@ -109,29 +58,26 @@ class NewVFIDataset(BaseVFIDataset):
 
 ```
 
-If you want create a dataset for a new low level CV task (e.g. denoise, derain, defog, and de-reflection), you can inheriting from [BaseDataset](/mmedit/datasets/base_dataset.py).
+If you want create a dataset for a new low level CV task (e.g. denoise, derain, defog, and de-reflection), you can inherit from [BasicImageDataset](/mmedit/datasets/basic_image_dataset.py) or [BasicFramesDataset](/mmedit/datasets/basic_frames_dataset.py).
 
-Here is an example of create a base dataset for denoising:
+Here is an example of creating a base dataset for denoising:
```
Comment (plyfager) [inline docs/en/advanced_guides/dataset.md:63]: please check whether the code of ``BaseDnDataset`` is correct.

File: docs/en/advanced_guides/transforms.md [hunk 3]
```diff
@@ -290,33 +244,25 @@ For each operation, we list the related dict fields that are added/updated/remov
...
 
 ```python
 import random
-from mmdet.datasets import PIPELINES
+from mmedit.registry import TRANSFORMS
 
 
-@PIPELINES.register_module()
+@TRANSFORMS.register_module()
 class MyTransform:
```
Comment (plyfager) [inline docs/en/advanced_guides/transforms.md:266]: Transforms are supposed to inherit from ``BaseTransform``.

Comment (plyfager) [reply docs/en/advanced_guides/transforms.md:266]: Also check the demo code below. Example: ``__call__`` --> ``transform``.

File: docs/en/migration.md [hunk 4]
```diff
@@ -1 +1,511 @@
-# Migration 
\ No newline at end of file
+# Migration from MMEdit 0.x
+
+We introduce some modifications in MMEdit 1.x. This document will help users of the 0.x version to quickly migrate projects to the newest version.
+
+## New dependencies
+
+MMEdit 1.x depends on some new packages, you can prepare a new clean environment and install again according to the install tutorial. Or install the below packages manually.
```
Comment (plyfager) [inline docs/en/migration.md:7]: attach tutorial url

File: docs/en/migration.md [hunk 5]
```diff
@@ -1 +1,511 @@
...
+
+- MMEngine: MMEngine is the core the OpenMMLab 2.0 architecture, and we splited many compentents unrelated to computer vision from MMCV to MMEngine.
```
Comment (plyfager) [inline docs/en/migration.md:9]:
```suggestion
- MMEngine: MMEngine is the core of the OpenMMLab 2.0 architecture, and we splited many compentents unrelated to computer vision from MMCV to MMEngine.
```

File: docs/en/migration.md [hunk 6]
```diff
@@ -1 +1,511 @@
...
+- MMCV: The computer vision package of OpenMMLab. This is not a new dependency, but you need to upgrade it to above 2.0.0rc0 version.
+
+## Overall Structures
+
+We refactor overall structures in MMEdit 1.x as following.
+
+- The  `core` in the old versions of MMEdit is split into `engine`, `evaluation`, `structures`, and `visualization` in MMEdit 1.x.
```
Comment (plyfager) [inline docs/en/migration.md:16]:
```suggestion
- The  `core` in the old versions of MMEdit is split into `engine`, `evaluation`, `structures`, and `visualization`.
```

File: docs/en/migration.md [hunk 7]
```diff
@@ -1 +1,511 @@
...
+
+We rename config file to new template: `{model_settings}_{module_setting}_{training_setting}_{datasets_info}`.
+
+### Model settings
+
+We update model settings in MMEdit 1.x. Important modifications are as following.
+
+- Remove `pretrained` fields.
+- Add `train_cfg` and `test_cfg` fields in model settings.
+- Add `data_preprocessor` fields. Normalization and color space transforms operations are moved from datasets transforms pipelines to data_preprocessor.
```
Comment (plyfager) [inline docs/en/migration.md:32]: tell users we will introduce data_preprocessor later (or in models.md).

File: docs/en/advanced_guides/transforms.md [hunk 8]
```diff
@@ -10,70 +10,41 @@ A pipeline consists of a sequence of operations. Each operation takes a dict as
...
-    dict(type='Flip', keys=['lq', 'gt'], flip_ratio=0.5, direction='vertical'),
-    dict(type='RandomTransposeHW', keys=['lq', 'gt'], transpose_ratio=0.5),
-    dict(type='MirrorSequence', keys=['lq', 'gt']),
-    dict(type='FramesToTensor', keys=['lq', 'gt']),
-    dict(type='Collect', keys=['lq', 'gt'], meta_keys=['lq_path', 'gt_path'])
+    dict(
+        type='Flip', keys=['img', 'gt'], flip_ratio=0.5, direction='vertical'),
+    dict(type='RandomTransposeHW', keys=['img', 'gt'], transpose_ratio=0.5),
+    dict(type='MirrorSequence', keys=['img', 'gt']),
+    dict(type='ToTensor', keys=['img', 'gt']),
```
Comment (plyfager) [inline docs/en/advanced_guides/transforms.md:30]: ``ToTensor`` may be removed.

File: docs/en/advanced_guides/transforms.md [hunk 9]
```diff
@@ -10,70 +10,41 @@ A pipeline consists of a sequence of operations. Each operation takes a dict as
...
-        channel_order='rgb'),
-    dict(type='RescaleToZeroOne', keys=['lq', 'gt']),
-    dict(type='FramesToTensor', keys=['lq', 'gt']),
-    dict(
-        type='Collect',
-        keys=['lq', 'gt'],
-        meta_keys=['lq_path', 'gt_path', 'key'])
+    dict(type='LoadImageFromFile', key='img', channel_order='rgb'),
+    dict(type='LoadImageFromFile', key='gt', channel_order='rgb'),
+    dict(type='ToTensor', keys=['img', 'gt']),
```
Comment (plyfager) [inline docs/en/advanced_guides/transforms.md:38]: mentioned above

File: docs/en/advanced_guides/transforms.md [hunk 10]
```diff
@@ -10,70 +10,41 @@ A pipeline consists of a sequence of operations. Each operation takes a dict as
...
-    dict(type='MirrorSequence', keys=['lq']),
-    dict(type='FramesToTensor', keys=['lq', 'gt']),
-    dict(
-        type='Collect',
-        keys=['lq', 'gt'],
-        meta_keys=['lq_path', 'gt_path', 'key'])
+    dict(type='LoadImageFromFile', key='img', channel_order='rgb'),
+    dict(type='LoadImageFromFile', key='gt', channel_order='rgb'),
+    dict(type='MirrorSequence', keys=['img']),
+    dict(type='ToTensor', keys=['img', 'gt']),
```
Comment (plyfager) [inline docs/en/advanced_guides/transforms.md:46]: mentioned above

File: docs/en/migration.md [hunk 11]
```diff
@@ -1 +1,511 @@
...
+        direction='horizontal'),  # Flip direction
+    dict(type='Flip',  # Flip images
+        keys=['lq', 'gt'],  # Images to be flipped
+        flip_ratio=0.5,  # Flip ratio
+        direction='vertical'),  # Flip direction
+    dict(type='RandomTransposeHW',  # Random transpose h and w for images
+        keys=['lq', 'gt'],  # Images to be transposed
+        transpose_ratio=0.5  # Transpose ratio
+        ),
+    dict(type='ToTensor', keys=['img', 'gt']),  # Convert images to tensor
```
Comment (plyfager) [inline docs/en/migration.md:201]: Remove ``ToTensor``.

File: docs/en/migration.md [hunk 12]
```diff
@@ -1 +1,511 @@
...
+    save_image=True,  # Save images during evaluation
+    gpu_collect=True)  # Use gpu collect
+```
+
+</td>
+
+<td valign="top">
+
+```python
+val_evaluator = [
```
Comment (plyfager) [inline docs/en/migration.md:352]: The old and new code examples should be identical in function and meaning. For example, if there is a metric in the left, there should be only one metric in the new code. You may check if other examples also have this problem

File: docs/en/migration.md [hunk 13]
```diff
@@ -1 +1,511 @@
...
+</thead>
+</table>
+
+### Schedule settings
+
+We update schedule settings in MMEdit 1.x. Important modifications are as following.
+
+- Now we use `optim_wrapper` field to specify all configuration about the optimization process. And the `optimizer` is a sub field of `optim_wrapper` now.
+- The `lr_config` field is removed and we use new `param_scheduler` to replace it.
+- The `total_iters` field is moved to `train_cfg`, `val_cfg` and `test_cfg`, which configure the loop in training, validation and test.
```
Comment (plyfager) [inline docs/en/migration.md:377]:
```suggestion
- The `total_iters` field is moved to `train_cfg` as `max_iters`, `val_cfg` and `test_cfg`, which configure the loop in training, validation and test.
```

File: docs/en/migration.md [hunk 14]
```diff
@@ -1 +1,511 @@
...
+resume = False  # Resume checkpoints from a given path, the training will be resumed from the epoch when the checkpoint's is saved.
+```
+
+</td>
+
+</tr>
+</thead>
+</table>
+
+### Others
```
Comment (plyfager) [inline docs/en/migration.md:500]: ## Config

File: docs/en/migration.md [hunk 15]
```diff
@@ -1 +1,511 @@
...
+
+More details of config are shown in [config guides](/docs/en/user_guides/config).
+
+## Model
```
Comment (plyfager) [inline docs/en/migration.md:504]: ### Model

File: docs/en/migration.md [hunk 16]
```diff
@@ -1 +1,511 @@
...
+test_cfg = dict(type='TestLoop')  # The name of test loop type
+```
+
+</td>
+
+</tr>
+</thead>
+</table>
+
+### Schedule settings
```
Comment (plyfager) [inline docs/en/migration.md:371]: You may also attach https://github.com/open-mmlab/mmengine/blob/main/docs/zh_cn/migration/migrate_param_scheduler_from_mmcv.md here.

File: docs/en/user_guides/visualization.md [hunk 17]
```diff
@@ -1 +1,34 @@
...
+
+Here are configs using `LocalVisBackend` to save visual results as an example.
+
+```python
+vis_backends = [dict(type='LocalVisBackend')] # The type of visual backend
+visualizer = dict( # Config to build visualizer
+    type='ConcatImageVisualizer',
+    vis_backends=vis_backends,
+    fn_key='gt_path',
+    img_keys=['gt_img', 'input', 'pred_img'],
```
Comment (plyfager) [inline docs/en/user_guides/visualization.md:16]: may tell users what the meaning of ``fn_key``, ``img_keys`` etc.

File: docs/en/user_guides/visualization.md [hunk 18]
```diff
@@ -1 +1,34 @@
-# visualization 
\ No newline at end of file
+# Visualization
+
+Using `visualizer` in config file can save visual results when training or testing.
+MMEditing supports `Loacl` visual backend and `Tensorboard ` visual backend now.
+
+## LocalVisBackend
+
+Here are configs using `LocalVisBackend` to save visual results as an example.
```
Comment (plyfager) [inline docs/en/user_guides/visualization.md:8]: Here, you may briefly tell users what are passed to visualizer, and they should set which fields to save visual results.

File: docs/en/user_guides/visualization.md [hunk 19]
```diff
@@ -1 +1,34 @@
...
+```python
+vis_backends = [dict(type='TensorboardVisBackend')] # The type of visual backend
+visualizer = dict( # Config to build visualizer
+    type='ConcatImageVisualizer',
+    vis_backends=vis_backends,
+    fn_key='gt_path',
+    img_keys=['gt_img', 'input', 'pred_img'],
+    bgr2rgb=True)
+custom_hooks = [dict(type='BasicVisualizationHook', interval=1)] # Config of visualization hook
+```
```
Comment (plyfager) [inline docs/en/user_guides/visualization.md:34]: What are supposed to do then should be told. Where to see the result?

File: docs/en/user_guides/visualization.md [hunk 20]
```diff
@@ -1 +1,34 @@
...
+```python
+vis_backends = [dict(type='LocalVisBackend')] # The type of visual backend
+visualizer = dict( # Config to build visualizer
+    type='ConcatImageVisualizer',
+    vis_backends=vis_backends,
+    fn_key='gt_path',
+    img_keys=['gt_img', 'input', 'pred_img'],
+    bgr2rgb=True)
+custom_hooks = [dict(type='BasicVisualizationHook', interval=1)] # Config of visualization hook
+```
```
Comment (plyfager) [inline docs/en/user_guides/visualization.md:19]: Tell user where to see the result.

File: docs/en/user_guides/visualization.md [hunk 21]
```diff
@@ -1 +1,34 @@
-# visualization 
\ No newline at end of file
+# Visualization
+
+Using `visualizer` in config file can save visual results when training or testing.
+MMEditing supports `Loacl` visual backend and `Tensorboard ` visual backend now.
```
Comment (plyfager) [inline docs/en/user_guides/visualization.md:4]: Attach https://github.com/open-mmlab/mmengine/blob/main/docs/zh_cn/tutorials/visualization.md here.

**Review by plyfager — COMMENTED** [review]
you may just merge config_xxx.md into config.md. Then, Introduce the config system and give one example.

**Review by plyfager — COMMENTED** [review]
Since your document structure is deeper then base design. Maybe you need to check whether the structure of web page is the same with your expectation.

File: docs/en/migration.md [hunk 22]
```diff
@@ -1 +1,511 @@
...
+More details of config are shown in [config guides](/docs/en/user_guides/config).
+
+## Model
+
+We refactor models in MMEdit 1.x. Important modifications are as following.
+
+- The `models` in MMedit 1.x is refactored to five parts: `base_models`, `data_preprocessors`, `editors`, `layers` and `losses`.
+- Add `data_preprocessor` module in `models`. Normalization and color space transforms operations are moved from datasets transforms pipelines to data_preprocessor. The data out from the data pipeline is transformed by this module and then fed into the model.
+
+More details of config are shown in [model guides](/docs/en/advanced_guides/models).
```
Comment (plyfager) [inline docs/en/migration.md:511]:
```suggestion
More details of models are shown in [model guides](/docs/en/advanced_guides/models).
```

**Comment by zengyh1900** [comment]
faq.md is intended for  Frequently Asked Questions. 
Please refactor it by listing some questions and answers (You may list some popular issues here)

File: docs/en/notes/projects.md [hunk 23]
```diff
@@ -1 +1,30 @@
...
+- AdaInt:Learning Adaptive Intervals for 3D Lookup Tables on Real-time Image Enhancement, CVPR 2022 [\[paper\]](https://arxiv.org/abs/2204.13983)[\[github\]](https://github.com/ImCharlesY/AdaInt)
+- A New Dataset and Transformer for Stereoscopic Video Super-Resolution, CVPRW 2022 [\[paper\]](https://openaccess.thecvf.com/content/CVPR2022W/NTIRE/papers/Imani_A_New_Dataset_and_Transformer_for_Stereoscopic_Video_Super-Resolution_CVPRW_2022_paper.pdf)[\[github\]](https://github.com/H-deep/Trans-SVSR)
+- BasicVSR:The Search for Essential Components in Video Super-Resolution and Beyond, CVPR 2021 [\[paper\]](https://arxiv.org/abs/2012.02181)[\[github\]](https://github.com/ckkelvinchan/BasicVSR-IconVSR)
+- GLEAN:Generative Latent Bank for Large-Factor Image Super-Resolution, CVPR 2021 [\[paper\]](https://arxiv.org/abs/2012.00739)[\[github\]](https://github.com/ckkelvinchan/GLEAN)
+- DAN:Unfolding the Alternating Optimization for Blind Super Resolution, NeurIPS 2020 [\[paper\]](https://arxiv.org/abs/2010.02631v4)[\[github\]](https://github.com/AlexZou14/DAN-Basd-on-Openmmlab)
+
+## Other projects
```
Comment (zengyh1900) [inline docs/en/notes/projects.md:24]: ## Open-source projects

File: docs/en/advanced_guides/transforms.md [hunk 24]
```diff
@@ -10,70 +10,41 @@ A pipeline consists of a sequence of operations. Each operation takes a dict as
```
Comment (zengyh1900) [inline docs/en/advanced_guides/transforms.md:10]: For a long tutorial, give an outline at the beginning.

File: docs/en/advanced_guides/dataset.md [hunk 25]
```diff
@@ -2,85 +2,32 @@
```
Comment (zengyh1900) [inline docs/en/advanced_guides/dataset.md:2]: give an outline at the beginning.

File: docs/en/user_guides/visualization.md (hunk 21 above)
Comment (zengyh1900) [inline docs/en/user_guides/visualization.md:4]: typo: 'Local'

**Comment by zengyh1900** [comment]
There are multiple dataset-related files under user_guides/dataset. 
You can merge them into one if necessary. 
Besides, provide the commands or the link to tools for data preparation in the documentation of the dataset.

**Comment by zengyh1900** [comment]
About user_guides/config, in each documentation, you can divide the snippet of config code into multiple snippets and explain each snippet briefly. For example, dive a single config file into the parts of the model, train_dataloader, test_dataloader, etc.

File: docs/en/migration.md [hunk 26]
```diff
@@ -1 +1,511 @@
-# Migration 
\ No newline at end of file
+# Migration from MMEdit 0.x
+
```
Comment (zengyh1900) [inline docs/en/migration.md:2]: give an outline at the beginning.

File: docs/en/user_guides/datasets/dataset_prepare.md [hunk 27]
```diff
@@ -1,6 +1,23 @@
 # Dataset Preparation
```
Comment (zengyh1900) [inline docs/en/user_guides/datasets/dataset_prepare.md:2]: add an overall introduction, 
For example,
..... There are two ways to use datasets for training and testing models in MMEditing:
1. using downloaded datasets directly  
2. preprocess downloaded datasets before using them.

File: docs/en/user_guides/datasets/dataset_prepare.md [hunk 28]
```diff
@@ -1,6 +1,23 @@
...
+Some datasets need to be preprocessed before training or testing. We support many scripts to prepare datasets in [tools/data](/tools/data). And you can follow the tutorials of every dataset to run scripts.
+For example, we recommend to crop the DIV2K images to sub-images. We provide a script to prepare cropped DIV2K dataset. You can run following command:
+
+```shell
+python tools/data/super-resolution/div2k/preprocess_div2k_dataset.py --data-root ./data/DIV2K
+```
+
+We support detailed tutorials and split them according to different tasks. Please follow the corresponding tutorials for data preparation of different tasks.
 
 - [Prepare Inpainting Datasets](inpainting_datasets.md)
```
Comment (zengyh1900) [inline docs/en/user_guides/datasets/dataset_prepare.md:22]: add a tilte before the list of datasets
for example,
### The overview of the datasets in MMEditing

**Pull request merged**